### PR TITLE
Feature/log mail

### PR DESCRIPTION
**Pull request by yulyulyharuka**
loggin mail when success sending mail

### CODE REVIEW

File: api/mail.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+func (s *smtpServer) getAddress() string {
+	return s.Host + ":" + s.Port
+}
+
+func SendMailHandler(db string) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var req SmtpRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+		if err != nil {
+			fmt.Fprint(w, fmt.Sprintf("%v", err))
```
Comment (david1312) [inline api/mail.go:44]: mungkin kurang return supaya stop fungsinya kalo error

Comment (david1312) [inline api/mail.go:44]: kurang return maybe

**Review by david1312 — CHANGES_REQUESTED** [review]

File: api/mail.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+func LogMail(email string, db storage.Storage) error {
+	logMail := models.LogMail{
+		Email:  email,
+		SentAt: time.Now(),
+	}
+	err := db.Create(logMail)
+	if err != nil {
+		return err
+	}
```
Comment (polipopoliko) [inline api/mail.go]: not necessary

File: api/storage/storage_db.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+func (o database) Create(obj models.LogMail) error {
+	_, err := o.db.Exec("INSERT INTO log_mail(email, sent_at) VALUES ($1, $2);", obj.Email, obj.SentAt)
+	return err
+}
+
+func (o database) List() ([]models.LogMail, error) {
+	rows, err := o.db.Query("SELECT email, sent_at FROM log_mail LIMIT 10;")
+	if err != nil {
+		log.Fatal(err)
```
Comment (polipopoliko) [inline api/storage/storage_db.go]: return nil, err instead??

File: api/storage/storage_db.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	defer rows.Close()
+
+	var sentMails []models.LogMail
+	for rows.Next() {
+		var sentMail models.LogMail
+		err = rows.Scan(&sentMail.Email, &sentMail.SentAt)
+		if err != nil {
+			log.Fatal(err)
```
Comment (polipopoliko) [inline api/storage/storage_db.go]: return nil, err instead??

**Pull request merged**